.circleci/run_style_check.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+import argparse
+from git.repo import Repo
+import subprocess
+import os
+import sys
+
+
+def run_black_formatter(apply_diff):
+    """
+    Runs black on committed files for style checks.
+    """
+    # Populate file list from the latest commit.
+    stream = os.popen(
```

Comment:
`with os.popen(... ) as stream:`

Reply:
That said, you might consider using `subprocess.check_output("git diff-tree --no-commit-id --name-only --diff-filter=ACM -r HEAD", shell=True, text=True).splitlines()`.

Reply:
Yes, thanks! This simplifies it quite a bit :)